### ISSUE TITLE
How to create bundle from existing training code 

### ISSUE THREAD

**Issue by AHarouni** [Project-MONAI/MONAI#5457]
Bundles are great feature. However I already have my own training loop as I needed to have more control (optimize gpu utilization, etc) 

I am following same high level structure as pre transforms, infer, post transform.
How can I move this into a bundle so it would be used by model zoo, monai deploy and monai label? 

Requests/ Suggestion:
Have interfaces for pre, post and infer similar to monai label for [training](https://github.com/Project-MONAI/MONAILabel/blob/2443582a0d4284528fa71c9aeac78f8781d693a8/monailabel/tasks/train/basic_train.py#L310) 
and 
[inference  ](https://github.com/Project-MONAI/MONAILabel/blob/20608cc81922f6a3027fa25bde9c5964cb4dd74f/monailabel/tasks/infer/basic_infer.py#L41)

 so if I can implement it, then monai core can create a way to read it and bundle it.

**Comment by yiheng-wang-nv** [comment]
Hi @AHarouni , thanks for the interests to bundle. You can refer to this tutorial about how to build a bundle:

https://github.com/Project-MONAI/tutorials/blob/main/bundle/get_started.ipynb

**Comment by AHarouni** [comment]
Thanks @yiheng-wang-nv for your response. Yes I have seen and followed the notebook you referred to. 

However, I think you miss understood my question. I am not starting from scratch. I already have my training loop and already have a model. I am interested in creating a bundle to consume it in deploy or monai label.

Currently I need to find my pre and post transforms and the infer method then convert them to json then build the bundle. I assume you take the config and turn it back to python code. 

For monai label for example I simply implemented their interface I pointed to and I have my app running. so every time I have a new model or change the transform in my code, monai label gets pointed to the same file and picks it up.

I am requesting a similar thing to create a bundle. 
Please let me know if you get my point or if my request makes sense.

**Comment by wyli** [comment]
python API added which should make this use case simpler https://github.com/Project-MONAI/tutorials/tree/main/bundle/python_bundle_workflow

**Comment by AHarouni** [comment]
Hi @wyli 
Thanks for following up on this. If I understand correctly I can simply implement this [interface](https://github.com/Project-MONAI/tutorials/blob/main/bundle/python_bundle_workflow/scripts/inference.py)
if so I can create a bundle, without using json or yaml files at all. 

I was hoping you can follow the interfaces which monai label is using but this is getting closer.
